src/lib/desmoContract-module.ts

```diff
@@ -180,21 +190,24 @@ export class DesmoContract {
     taskId: string;
     result: string;
   }> {
-    const myIexec = this.iexec as IExec;
     if (!this.isConnected) {
       throw new Error(
         'This method requires the wallet signer to be already signed-in!',
       );
     }
-    
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const self = this;
-    return new Promise( (resolve, reject) => {
-      (function loop(){
+    return new Promise((resolve/*, reject*/) => {
+      (function loop(): void {
         setTimeout(async () => {
+          if (self.iexec === undefined) {
+            throw Error('A connection to iExec is required!');

```

Comment:
You have to call reject here instead of throwing the Error

Reply:
Solved in the updated version of the code.